### PR TITLE
fix(encounter): preserve effect envelope metadata (#701)

### LINKED ISSUE #701: encounter: project effect-event correlation and occurrence time onto damage/status envelopes
## Goal

Preserve toolkit-authored encounter-event metadata when translating correlated damage and status effects onto the v1alpha2 wire.

This is the API-side prerequisite discovered while designing the client envelope passthrough in rpg-dnd5e-web#582 for live combat pacing (#581).

## Verified current state

The toolkit already does the correct work:

- `Encounter.publishAttackOutcome` creates one correlation id and publishes `ActionResolvedEvent`, `AttackResolvedEvent`, and hit-only `DamageDealtEvent` through `publishCorrelated`.
- `eventMeta` carries `CorrelationID()` and broker-authored `OccurredAt()` on every encounter event.
- `DamageDealtEvent` and `ConditionAppliedEvent` embed that same metadata.

The API translation drops it for effect events:

- `translateActionResolvedEvent` and `translateAttackResolvedEvent` project `CorrelationId` and `OccurredAt()` correctly.
- `translateDamageDealtEvent` builds `EntityDamaged` without `CorrelationId` and stamps the supplied handler `now` instead of `e.OccurredAt()`.
- `translateConditionAppliedEvent` does the same for `StatusApplied`.

The result is a declared hit whose action and roll share a correlation id, but whose damage envelope arrives with an empty id. A client cannot reliably tie the damage amount to the correct roll during overlapping, opportunity, or repeat attacks.

## Scope

- Project `string(e.CorrelationID())` onto `EncounterEvent.CorrelationId` in `translateDamageDealtEvent` and `translateConditionAppliedEvent`.
- Project `e.OccurredAt()` onto each envelope timestamp instead of substituting the handler clock.
- Pass empty correlation ids through unchanged for genuinely uncorrelated events.
- Add translator tests proving exact non-empty correlation and occurrence time passthrough for both event types.
- Add an attack-chain assertion proving `ActionResolved`, `AttackResolved`, and hit-only `EntityDamaged` can carry the same correlation id end to end on the wire.

## Out of scope

- Proto or toolkit changes; both fields already exist and are populated upstream.
- Client callback changes; those are rpg-dnd5e-web#582.
- Combat-pacing grouping, completion, animation, or UI behavior; those remain rpg-dnd5e-web#581.
- Inventing correlation for events the toolkit leaves uncorrelated.

## Done bar

A correlated toolkit damage/status effect reaches the v1alpha2 stream with its exact toolkit correlation id and occurrence timestamp. Existing visibility projection and payload fields remain unchanged, targeted translator tests pass, and full `make ci-check` passes.

**Blocks:** KirkDiggler/rpg-dnd5e-web#581
**Related:** KirkDiggler/rpg-dnd5e-web#582
**Team:** Platform
**Feature:** Game Screen

---
— asset-pipeline agent, on behalf of KirkDiggler

### PR DESCRIPTION
**Pull request by KirkDiggler**
Closes #701. Relates to KirkDiggler/rpg-dnd5e-web#582; both are required before KirkDiggler/rpg-dnd5e-web#581.

Canonical design/plan: KirkDiggler/rpg-project#118.

Projects `DamageDealtEvent` and `ConditionAppliedEvent` `OccurredAt()` and `CorrelationID()` values onto the existing `EntityDamaged` and `StatusApplied` envelope fields. No proto, toolkit, or web changes.

TDD evidence: the focused metadata suite was RED before the change and GREEN after; the full repository test phase passes. Coverage includes empty-correlation preservation and the declared-hit chain correlation/timestamp test.

`make ci-check` remains NON-GREEN on current `origin/main` due to pre-existing mock-generation drift (7 listed mocks) and 29 lint findings outside the changed files; this PR does not claim CI passed. Fresh verification confirmed no finding in the changed files.

— asset-pipeline agent, on behalf of KirkDiggler

### CODE REVIEW

File: internal/handlers/dnd5e/v2/encounter/translate.go [hunk 1]
```diff
@@ -712,8 +712,9 @@ func translateDamageDealtEvent(e *events.DamageDealtEvent, viewer core.PlayerID,
 		}
 	}
 	return &encounterv2pb.EncounterEvent{
-		Sequence:  int64(e.Sequence()),
-		Timestamp: timestamppb.New(now),
+		Sequence:      int64(e.Sequence()),
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/handlers/dnd5e/v2/encounter/translate.go]: Fixed in 76af463: added `//nolint:gosec // sequence is monotonic; fits int64` to the EntityDamaged envelope sequence conversion. — asset-pipeline agent, on behalf of KirkDiggler

File: internal/handlers/dnd5e/v2/encounter/translate.go [hunk 2]
```diff
@@ -750,8 +751,9 @@ func translateConditionAppliedEvent(e *events.ConditionAppliedEvent, viewer core
 		out.SourceEntityId = &s
 	}
 	return &encounterv2pb.EncounterEvent{
-		Sequence:  int64(e.Sequence()),
-		Timestamp: timestamppb.New(now),
+		Sequence:      int64(e.Sequence()),
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/handlers/dnd5e/v2/encounter/translate.go]: Fixed in 76af463: added `//nolint:gosec // sequence is monotonic; fits int64` to the StatusApplied envelope sequence conversion. — asset-pipeline agent, on behalf of KirkDiggler

File: internal/handlers/dnd5e/v2/encounter/translate_combat_test.go [hunk 3]
```diff
@@ -563,6 +585,36 @@ func (s *TranslateSuite) TestTranslateEvent_ConditionAppliedEvent_HappyPath_Full
...
+	gameTime := time.Date(2026, 7, 24, 10, 11, 13, 456789000, time.UTC)
+	evt := events.NewConditionAppliedEvent("enc-1", uint64(72), "goblin-1", "char-A", "dnd5e:conditions:poisoned", 3,
+		map[core.PlayerID]events.ConditionAppliedSlice{"player-A": {Visible: true}})
+	evt.Stamp(gameTime, "corr-701-effect")
+
+	out, err := v2encounter.TranslateEvent(evt, "player-A", s.now)
+	s.Require().NoError(err)
+	s.Require().Equal("corr-701-effect", out.GetCorrelationId())
+	s.Require().Equal(gameTime, out.GetTimestamp().AsTime())
+}
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply internal/handlers/dnd5e/v2/encounter/translate_combat_test.go:598]: Fixed in 76af463: added explicit ConditionApplied empty-correlation passthrough coverage and aligned the stamped condition test constructor to target char-A from source goblin-1. — asset-pipeline agent, on behalf of KirkDiggler

**Comment by KirkDiggler** [comment]
## GATE REVIEW

**Verdict: MERGE-READY**

Highlights:
- Independent base-to-head review confirms toolkit-authored `OccurredAt()` and `CorrelationID()` are projected verbatim onto both `EntityDamaged` and `StatusApplied` envelopes.
- Scope remains API-only: two translator paths plus focused tests; no proto, toolkit, web, payload, visibility, or public API changes.
- Regression coverage proves non-empty and empty correlation passthrough for both effect types and a declared action -> attack -> damage chain sharing one correlation.
- All three Copilot inline findings were addressed and acknowledged individually.
- Current GitHub `build` and `test` checks pass on head `76af463`.

Findings: none.

Residual boundary: this PR is one of two prerequisites for web#581; web#582 must also merge before UI consumes correlated effect metadata.

— asset-pipeline agent, on behalf of KirkDiggler
